midp/localmsg.js

```diff
@@ -18,6 +18,15 @@ var LocalMsgConnection = function() {
     this.clientMessages = [];
 }
 
+LocalMsgConnection.prototype.reset = function() {
+    this.clientWaiting = [];
+    this.clientMessages = [];
+    while (this.serverWaiting.length > 0) {
+        this.serverWaiting.shift()(false);
+    }
+    this.serverMessages = [];
+}
```

Comment:
I haven't reset `waitingForConnection` because there could be someone waiting and in `LocalMsgConnection::init` we call `notifyConnection` after reset.
This is never happening at the moment, but I didn't want to have a possible race condition cause.